### ISSUE TITLE
Is there an option to use nn.DataParallel rather than nn.DistributedDataParallel?

### ISSUE THREAD

**Issue by AntreasAntoniou** [huggingface/accelerate#975]
### System Info

```Shell
- `Accelerate` version: 0.15.0
- Platform: Linux-4.19.0-23-cloud-amd64-x86_64-with-glibc2.31
- Python version: 3.10.8
- Numpy version: 1.23.5
- PyTorch version (GPU?): 1.13.1 (True)
- `Accelerate` default config:
        - compute_environment: LOCAL_MACHINE
        - distributed_type: MULTI_GPU
        - mixed_precision: no
        - use_cpu: False
        - dynamo_backend: NO
        - num_processes: 2
        - machine_rank: 0
        - num_machines: 1
        - gpu_ids: all
        - main_process_ip: None
        - main_process_port: None
        - rdzv_backend: static
        - same_network: True
        - main_training_function: main
        - deepspeed_config: {}
        - fsdp_config: {}
        - megatron_lm_config: {}
        - downcast_bf16: no
        - tpu_name: None
        - tpu_zone: None
        - command_file: None
        - commands: None
```

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [X] My own task or dataset (give details below)

### Reproduction

I am currently trying to train a large contrastive model and I require a way to allow say to use 8 x GPUs and each having 128 items on each one processed, but then at the end doing the (8 x 128)**2 dot product before returning the loss. That essentially means that at some point before gradient computation I need to send the results from all my models to one place. 

I am able to do this with nn.DataParallel and I was wondering whether accelerate has any way to use nn.DataParallel instead of nn.DistributedDataParallel? 

Alternatively are there any other ways that are more efficient other than nn.DataParallel that I should be using?

### Expected behavior

```Shell
An explicit option to use nn.DataParallel rather than nn.DistributedDataParallel in the config.
```

**Comment by sgugger** [comment]
No, Accelerate does not provide options to use DataParallel as PyTorch does not recommend its use.

**Comment by AntreasAntoniou** [comment]
Cool. Do you have any advice on how to go about breaking a set of images in N parts, sending each part to a separate GPU and then computing the final loss metric on a concatenated tensor made of all models' N outputs?

**Comment by sgugger** [comment]
You can use `accelerate.gather` to concatenate a tensor across GPUs.

**Comment by AntreasAntoniou** [comment]
@sgugger Thanks for the response! I'll try this out and report with a piece of code that worked for me, for anyone who would like to do the same in the future.

**Comment by AntreasAntoniou** [comment]
OK. I can confirm that this suggestion works. Should I make a note of this in the docs and make a PR? Is that OK?

**Comment by partha117** [comment]
> OK. I can confirm that this suggestion works. Should I make a note of this in the docs and make a PR? Is that OK?

Can you post your solution? I am facing the same issue. However, `accelerate.gather` does not retain the gradient in my case. How did you achieve it?
